packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -316,6 +316,7 @@ export function RichTextWrapper(
 						transformation.transform( { content: value.text } ),
 					] );
 					__unstableMarkAutomaticChange();
+					return;
```

Comment:
This leads to the `onCustomEnter` handler _not_ being called, but that should be OK. If I understand correctly, it's used to handle things like list splitting (#43949).

Reply:
The #43949 test plan succeeds for me on an iPhone 15 Pro simulator.